### DISCUSSION TITLE
How to trigger submission despite invalid fields?

### DISCUSSION THREAD

**Discussion by debel27** [react-hook-form/react-hook-form#9852] [Q&A] [answered]
I'm tasked to implement a form with auto-save behavior. One strong requirement is that invalid fields should not prevent valid fields from being auto-saved.

The way I see it, auto-save should be implemented by triggering a form submit whenever a field value changes (with possibly some debouncing). However, invalid fields happen block the whole submit operation until the corresponding errors are fixed. As such, an invalid form value prevents valid changes from being submitted. It's a "all or nothing" situation.

Is there a way to implement my use case with react-hook-form?

**Comment by Moshyfawn** [comment, marked as answer]
The submission validation happens inside `handleSubmit`. Pass your auto-save method to the event in question without putting it inside `handleSubmit`

**Reply by debel27** [reply]
Thanks for your reply. 
If I understand correctly, you're suggesting to call `onSubmit` directly instead of `handleSubmit(onSubmit)()` in case of auto-save. In that case, how can I distinguish valid fields from invalid ones?

**Reply by Moshyfawn** [reply]
In this case, you can pass your `auto-save` method to the second `handleSubmit` callback
```js
handleSubmit(onSubmit, onSubmitDraft)
```

You'll get the `errors` object `onSubmitDraft`.
To get the form data, you can call `getValues` inside `onSubmitDraft`
```js
const { getValues } = useForm()

const onSubmitDraft = errors => {
  const values = getValues()
}
```

**Reply by debel27** [reply]
This helps. Thanks a lot!
